pyext/fastfunc.c

```diff
@@ -104,6 +102,14 @@ func_Utf8DecodeOne(PyObject *self, PyObject *args) {
     codepoint_or_error = decode_result.codepoint;
   }
 
+  // utf8_decode treats zero-bytes as C-style NUL-terminators. But python2/oils
+  // strings treat these as zero-codepoints. Translate END_OF_STREAM errors
+  // (resulting from zero-bytes) to valid zero-codepoints.
+  if (decode_result.error == UTF8_ERR_END_OF_STREAM) {
```

Comment:
Hm it seems like this works but isn't it cleaner to not return UTF8_ERR_END_OF_STREAM in the first place?

Basically I imagine



- `0xce 0x00` -- truncated encoding relies NUL bytes, returns  UTF8_ERR_TRUNCATED_BYTES
- 0x00` after some other valid encoding -- this is UTF8_OK 

I think it's possible to distinguish these cases in utf8decode()?  Or would it also need a length?